pkg/controller/deployment/deployment_controller.go

```diff
@@ -102,36 +332,46 @@ func (d *DeploymentController) reconcileRollingUpdateDeployment(deployment exten
 	allRCs := append(oldRCs, newRC)
 
 	// Scale up, if we can.
-	scaledUp, err := d.reconcileNewRC(allRCs, newRC, deployment)
+	scaledUp, err := dc.reconcileNewRC(allRCs, newRC, deployment)
 	if err != nil {
 		return err
 	}
 	if scaledUp {
 		// Update DeploymentStatus
-		return d.updateDeploymentStatus(allRCs, newRC, deployment)
+		return dc.updateDeploymentStatus(allRCs, newRC, deployment)
 	}
 
 	// Scale down, if we can.
-	scaledDown, err := d.reconcileOldRCs(allRCs, oldRCs, newRC, deployment)
+	scaledDown, err := dc.reconcileOldRCs(allRCs, oldRCs, newRC, deployment)
 	if err != nil {
 		return err
 	}
 	if scaledDown {
 		// Update DeploymentStatus
-		return d.updateDeploymentStatus(allRCs, newRC, deployment)
+		return dc.updateDeploymentStatus(allRCs, newRC, deployment)
 	}
 	// TODO: raise an event, neither scaled up nor down.
```

Comment:
Note that we should call updateDeploymentStatus only when the condition I stated is correct.
Before your change, updateDeploymentStatus was only called when some RCs were scaled up or down and hence it always calls client.Deployment.UpdateStatus() without checking if there is any change in status.